frontend/stores/user.js

```diff
@@ -0,0 +1,207 @@
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                method: 'POST',
+                body
+            }
+
+            const { data: register, pending, error } = await useAsyncData('register', () => $cmsApi('/api/auth/register/', config))
+
+            if (error.value) {
+                let warnMessage = null
+                for (const [key, value] of Object.entries(error._value.data)) {
+                    warnMessage = warnMessage ? `${warnMessage} \n\n ${key}: ${value}` : `${key}: ${value}`
+                }
+                this.register.succes = false
+                // Notification
+                global.negativeNotify(warnMessage)
+
+            }
+            if (register?.value) {
+                this.register.succes = true
+                // Notification
+                global.succesNotify('Register complete')
+                await navigateTo('/login')
+            }
+
+        },
+        /**
+         * Login user
+         * @param {*} email 
+         * @param {*} password 
+         * TODO: add pending functionality
+         */
+        async loginUser(
+            email,
+            password,
+        ) {
+            const global = useGlobalStore()
+            this.userData.error = ''
+            this.userData.pending = true
+
+            const config = {
+                method: 'POST',
+                body: {
+                    email,
+                    password,
+                },
+            }
+
+            const { data: login, pending, error } = await useAsyncData('login', () => $cmsApi('/api/auth/login/', config))
+
+            if (error.value) {
+                console.log('error: ', error.value)
+                let warnMessage = null
+                for (const [key, value] of Object.entries(error._value.data)) {
+                    warnMessage = warnMessage ? `${warnMessage} \n\n ${key}: ${value}` : `${key}: ${value}`
+                }
+
+                // this.userData.error = err
+                this.userData.isAuthenticated = false
+
+                // Notification
+                global.negativeNotify(warnMessage)
+
+            }
+            if (login?.value) {
+                console.log('login //> ', login.value)
+                this.userData = {
+                    ...this.userData,
+                    isAuthenticated: true,
+                    token: login?.value.token,
+                    user: login?.value.user,
+                    // ...res
+                }
+                localStorage.setItem('token', this.userData.token)
+                // Notification
+                global.succesNotify('Login complete')
+                // NICETOHAVE: It mees like with the route `redirectedFrom` api you can get the previous link, you can use this to pass in the navigateTo function
+                // See: https://nuxt.com/docs/api/composables/use-route
+                await navigateTo('/design')
```

Comment:
When logging in, after successful login, I am not redirected to my available surveys (design page). Only manually reloading the page allows me to go there.

On closer inspection, I see a few API requests to `/api/auth/me/` that are still `Unauthorized`, so it looks like the token received from login is not updated when other calls are made before the page is reloaded.
I guess an easy fix would be to just automatically reload the page after login? Is there a command for that?
![image](https://user-images.githubusercontent.com/52563982/214099613-10be172c-657e-45ed-9a20-bac46a2c6711.png)

![image](https://user-images.githubusercontent.com/52563982/214099590-7fa6e2dc-6350-41c3-8060-593306b35862.png)